src/air/constraints/evaluator.rs

```diff
@@ -171,6 +175,7 @@ impl<'poly, F: IsFFTField, A: AIR + AIR<Field = F>> ConstraintEvaluator<'poly, F
                 .map(|row| {
                     zerofier_evaluations
                         .iter()
+                        .cycle()
```

Comment:
I don't understand how this `cycle` works here or what works differently now. Could you explain a bit further?